numba_pandas/__init__.py

```diff
@@ -0,0 +1,11 @@
+
+def _init_extension():
```

Comment:
As far as I understand (may be I'm wrong) this code should be called from Numba or should be existed in Numba.
I don't think we need additional `__init__` for HPAT. I think we already have initialization procedure.

Reply:
1. I will explain how Numba calls this code. No magic.
In `setup.py` I have added an extension point:
```
      entry_points={
          "numba_extensions": [
              "init = numba_pandas:_init_extension",
          ],
      },
```
This extension point sets `init` to exactly this function.
Numba >=0.46 can find this extension point and call the function.
This is a standard Python packaging technique when one package access to plugins provided by other packages. Each package can publish extension points which are visible to other packages.
Numba states that this technique is supported in 0.46 - it is described in Numba documentation - http://numba.pydata.org/numba-doc/latest/extending/entrypoints.html.
2. I agree, initialization part should be revised. Will make issue and PR for it.
Quick thoughts:
Dedicated function `_init_extension()` makes it possible to control which parts of HPAT will be loaded for extending Numba. With `import hpat` the whole HPAT will be loaded.

Reply:
@PokhodenkoSA 
2. Thank you
1. Does it mean, the SDC (old HPAT) will initialized every time when Numba works? I thought it happen if Pandas package required only.

Reply:
1. SDC will be initializes every time when Numba is imported. We can control the part of SDC which will be loaded.
> I thought it happen if Pandas package required only.
Could you explain "required"? Does it mean "used in code - import pandas" or "Pandas is installed"?
We could check if Pandas is installed but Pandas is in requirements of SDC. Actually we could remove Pandas from requirements of SDC and load SDC parts depending on whether Pandas is installed.

Reply:
Can I keep this function `_init_extension()` in `numba_pandas/__init__.py`? Or should I move it to `hpat` (`sdc`) folder?

Reply:
I not sure but I would propose to move it into `hpat/__init__.py`